subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/StaleOutputIntegrationTest.groovy

```diff
@@ -168,19 +168,23 @@ class StaleOutputIntegrationTest extends AbstractIntegrationSpec {
         succeeds("myTask")
 
         when:
-        // Now that we produce this, we can detect the situation where
-        // someone builds with Gradle 4.3, then 4.2 and then 4.3 again.
-        file(".gradle/buildOutputCleanup/cache.properties").text = """
-            gradle.version=1.0
-        """
-        // recreate the output
+        changeGradleVersion()
         dirInBuildDir.createDir()
         staleFileInDir.touch()
         fileInBuildDir.touch()
         then:
         succeeds("myTask")
     }
 
+    // Now that we produce this, we can detect the situation where
+    // someone builds with Gradle 4.3, then 4.2 and then 4.3 again.
+    // recreate the output
+    private void changeGradleVersion() {
```

Comment:
Let's rename this method to something representing the goal to be achieved.

Reply:
Renamed to `invalidateBuildOutputCleanupState`.